enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,260 @@
+title: single-node-deployment-with-bootstrap-in-place
+authors:
+  - "@dhellmann"
+  - "@eranco"
+  - "@romfreiman"
+  - "@tsorya"
+reviewers:
+  - TBD
+  - "@mrunalp"
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-13
+last-updated: 2020-12-13
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/pull/560
+  - https://github.com/openshift/enhancements/pull/302
+---
+
+# Single Node deployment with bootstrap-in-place
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add the new [single-node production deployment](https://github.com/openshift/enhancements/pull/560/files)
+we need a way to install such cluster without an extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node OpenShift using liveCD that perform the bootstrap logic and reboots to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Resource constraints are the primary motivation for single-node OCP clusters, and removing the need for a
+separate bootstrap machine significantly reduces the resources required to install the cluster.
+
+Requiring the auxiliary node for installing a Single Node OpenShift brings to play some requirements:
+1. The obvious additional node.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
```

Comment:
How does the auxiliary node mean we cannot use baremetal IPI?

Reply:
Currently, BM ipi brings all the extra networking parts (keepalived, haproxy, mdns) that are irrelevant for a single node.
If we ignore that, I actually dont know whether the deployment will work. Dont forget that metal does not use rhcos based image, so not even sure what should be done. Anyhow, it's a non-goal for this enhancement.

Reply:
You are right it's out of context